cwl/src/main/scala/cwl/ontology/Schema.scala

```diff
@@ -0,0 +1,133 @@
+package cwl.ontology
+
+import cats.instances.list._
+import cats.syntax.traverse._
+import common.validation.ErrorOr._
+import common.validation.Validation._
+import cwl.ontology.Schema._
+import org.semanticweb.owlapi.apibinding.OWLManager
+import org.semanticweb.owlapi.model._
+import org.semanticweb.owlapi.reasoner.structural.StructuralReasonerFactory
+import org.semanticweb.owlapi.reasoner.{OWLReasoner, OWLReasonerFactory}
+import org.semanticweb.owlapi.util.OWLAPIStreamUtils
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+/**
+  * OWL/RDF Schema lookup.
+  *
+  * @param schemaIris IRI paths to OWL/RDF schemas.
+  * @param namespaces Additional/Override namespace prefixes.
+  */
+case class Schema(schemaIris: Seq[String], namespaces: Map[String, String]) {
+
+  /**
+    * Returns a full IRI based on a full or abbreviated IRI.
+    *
+    * A full IRI wrapped in < and > will not be checked for abbreviations but will be returned without the wrapper.
+    *
+    * @see https://www.w3.org/TR/owl2-syntax/#IRIs
+    */
+  def fullIri(name: String): String = getIri(name).getIRIString
+
+  /**
+    * Returns true if child is an equivalent of ancestor, and if not then recursively checks if any of the child's
+    * super classes are an equivalent of ancestor.
+    */
+  def isSubClass(child: String, ancestor: String): Boolean = {
+    val childIri: IRI = getIri(child)
+    val ancestorIri: IRI = getIri(ancestor)
+    val childClass: OWLClass = dataFactory.getOWLClass(childIri)
+    val ancestorClass: OWLClass = dataFactory.getOWLClass(ancestorIri)
+    val schemaReasoner: OWLReasoner = reasonerFactory.createReasoner(schemaOntology)
+    try {
+      Schema.isSubClass(schemaReasoner, childClass, ancestorClass)
+    } finally {
+      schemaReasoner.dispose()
+    }
+  }
+
+  private val ontologyManager: OWLOntologyManager = OWLManager.createOWLOntologyManager
+  private val dataFactory: OWLDataFactory = ontologyManager.getOWLDataFactory
+  private val reasonerFactory: OWLReasonerFactory = new StructuralReasonerFactory
+  private val schemaOntology: OWLOntology = ontologyManager.createOntology()
+  private val schemaPrefixManager: PrefixManager =
```

Comment:
can `OWLOntologyManager` and `OWLReasonerFactory` be constructor params w/ defaults?

Reply:
Exposed! Though no-one is using those parameters afaict & in general I like to encapsulate APIs.

I'm really hoping **no one** touches the `owl-api` directly outside of this package. Depending on what happens in the future we may need to switch libraries, like we have many other times:
- `spray` ➔ `akka-http`
- `spray-json` ➔ `circe`
- `nashorn` ➔ `rhino`

For example, there are _still_ `.ask()` calls left over after Akka abandoned `spray` and we had to (mostly) upgrade.